src/canister/user_index/src/api/user_record/get_requester_principals_canister_id_create_if_not_exists_and_optionally_allow_referrer.rs

```diff
@@ -117,33 +119,62 @@ async fn new_user_signup(user_id: Principal) -> Result<Principal, String> {
     }
 
     let individual_user_template_canister_wasm = CANISTER_DATA.with_borrow(|canister_data| canister_data.wasms.get(&WasmType::IndividualUserWasm).unwrap().clone());
-    ic_cdk::spawn( async move {
-        // provision new canisters on the subnet if required.
-        let provision_canister_cnt = u64::min(INDIVIDUAL_USER_CANISTER_SUBNET_BATCH_SIZE, INDIVIDUAL_USER_CANISTER_SUBNET_MAX_CAPACITY - available_individual_user_canisters_cnt - individual_user_canisters_cnt);
-        if individual_user_canisters_cnt + available_individual_user_canisters_cnt < INDIVIDUAL_USER_CANISTER_SUBNET_MAX_CAPACITY && available_individual_user_canisters_cnt < INDIVIDUAL_USER_CANISTER_SUBNET_THREESHOLD {
-            provision_new_canister(provision_canister_cnt, individual_user_template_canister_wasm).await
-        }
-     });
+   
+    if individual_user_canisters_cnt < INDIVIDUAL_USER_CANISTER_SUBNET_THREESHOLD {
+        let new_canister_cnt = INDIVIDUAL_USER_CANISTER_SUBNET_BATCH_SIZE.min(backup_individual_user_canister_cnt);
+        ic_cdk::spawn(provision_new_available_canisters(new_canister_cnt, individual_user_template_canister_wasm));
+    }
+
+    if total_canister_provisioned_on_subnet < INDIVIDUAL_USER_CANISTER_SUBNET_MAX_CAPACITY && backup_individual_user_canister_cnt < BACKUP_INDIVIDUAL_USER_CANISTER_THRESHOLD {
+        let new_canister_cnt = BACKUP_INDIVIDUAL_USER_CANISTER_BATCH_SIZE.min(INDIVIDUAL_USER_CANISTER_SUBNET_MAX_CAPACITY - total_canister_provisioned_on_subnet);
+        ic_cdk::spawn(provision_new_backup_canisters(new_canister_cnt));
+    }
+
+
+   
+     
 
     response
 }
 
-async fn provision_new_canister(provision_canister_cnt: u64, individual_user_template_canister_wasm: CanisterWasm) {
+
+
+
+async fn provision_new_available_canisters(cnt: u64, individual_user_template_canister_wasm: CanisterWasm) {
+    let install_canister_wasm_futures = CANISTER_DATA.with_borrow(|canister_data| {
+        let mut backup_pool_canister = canister_data.backup_canister_pool.clone().into_iter();
+        (0..cnt).map(move |_| {
+            let canister_id = backup_pool_canister.next().unwrap();
+            let future = install_canister_wasm(canister_id, None, individual_user_template_canister_wasm.version.clone(), individual_user_template_canister_wasm.wasm_blob.clone());
+            future
+        })
+    });
+
+    let result_callback = |canister_id: Principal| {
+        CANISTER_DATA.with_borrow_mut(|canister_data| {
+            canister_data.available_canisters.insert(canister_id);
+            canister_data.backup_canister_pool.remove(&canister_id);
+        })
+    };
+    
+    let breaking_condition = || { CANISTER_DATA.with_borrow(|canister_data| canister_data.available_canisters.len() as u64 >= cnt) };
+
+    run_task_concurrently(install_canister_wasm_futures, 10, result_callback, breaking_condition).await;
+}
+
+
+async fn provision_new_backup_canisters(provision_canister_cnt: u64) {
```

Comment:
Minor nitpick but cnt vs count is hardly two keystrokes which autocomplete is going to fill anyway. Might as well use complete word variable names? 😅

Reply:
yes, will fix it